src/compositemark/boxplot.ts

```diff
@@ -2,86 +2,66 @@ import {isNumber} from 'vega-util';
 import {Channel} from '../channel';
 import {Config} from '../config';
 import {reduce} from '../encoding';
+import {GenericMarkDef, isMarkDef, MarkConfig} from '../mark';
 import {AggregatedFieldDef, BinTransform, CalculateTransform, TimeUnitTransform} from '../transform';
+import {Flag, keys} from '../util';
 import {Encoding, forEach} from './../encoding';
 import {Field, FieldDef, isContinuous, isFieldDef, PositionFieldDef, vgField} from './../fielddef';
 import * as log from './../log';
-import {MarkConfig} from './../mark';
 import {GenericUnitSpec, LayerSpec} from './../spec';
 import {Orient} from './../vega.schema';
-import {getMarkSpecificConfigMixins} from './common';
+import {getMarkDefMixins} from './common';
 
 
-export const BOXPLOT: 'box-plot' = 'box-plot';
-export type BOXPLOT = typeof BOXPLOT;
-export type BoxPlotStyle = 'boxWhisker' | 'box' | 'boxMid';
+export const BOXPLOT: 'boxplot' = 'boxplot';
+export type BoxPlot = typeof BOXPLOT;
 
+export type BoxPlotPart = 'box' | 'mid' | 'whisker';
 
-export interface BoxPlotDef {
-  /**
-   * Type of the mark.  For box plots, this should always be `"box-plot"`.
-   * [boxplot](compositemark.html#boxplot)
-   */
-  type: BOXPLOT;
+const BOXPLOT_PART_INDEX: Flag<BoxPlotPart> = {
+  box: 1,
+  mid: 1,
```

Comment:
Anyone has any better name for the mid tick?  

I don't want to make it two words `mid-tick`, but `mid` as a part name is also not very satisfying.

Reply:
@mattwchun `center` is perhaps better?